mod_cluster/undertow/src/main/java/org/wildfly/mod_cluster/undertow/UndertowServer.java

```diff
@@ -49,8 +49,12 @@ public UndertowServer(UndertowService service, Connector connector) {
     @Override
     public Iterable<Engine> getEngines() {
         // Currently, the mod_cluster subsystem only supports the default server
-        org.wildfly.extension.undertow.Server defaultServer = this.service.getServers().stream().filter(server -> server.getName().equals(this.service.getDefaultServer())).findFirst().get();
-        return Collections.singleton(new UndertowEngine(defaultServer, this.service, this.connector));
+        for (org.wildfly.extension.undertow.Server server : service.getServers()) {
+            if (server.getName().equals(service.getDefaultServer())) {
+                return Collections.singleton(new UndertowEngine(server, service, connector));
+            }
+        }
+        throw new IllegalStateException();
```

Comment:
The original code actually throws `NoSuchElementException` but `IllegalStateException` is fine too since we don't check for it specifically.

https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#findFirst--
https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#get--